### PR TITLE
Pi-hole: Add DHCP support

### PR DESCRIPTION
**Pull request by 0x346e3730**
Add the possibility to pi-hole to act as a DHCP server

### CODE REVIEW

File: pi-hole/docker-compose.yml [hunk 1]
```diff
@@ -24,3 +25,5 @@ services:
     networks:
       default:
         ipv4_address: $APP_PI_HOLE_IP
+    cap_add:
+      - NET_ADMIN
```
Comment (nevets963) [inline pi-hole/docker-compose.yml:29]: Do you know why this capability needs to be added?

**Review by nevets963 — COMMENTED** [review]
@0x346e3730 Thanks for the PR! Have you managed to test this within your network?

File: pi-hole/docker-compose.yml (hunk 1 above)
Comment (0x346e3730) [reply pi-hole/docker-compose.yml:29]: Yes, It's documented on https://github.com/pi-hole/docker-pi-hole as I explained in https://github.com/getumbrel/umbrel-apps/issues/145.

**Comment by 0x346e3730** [comment]
> @0x346e3730 Thanks for the PR! Have you managed to test this within your network?

I have not no as I have no test environment and cannot break my install

**Comment by nevets963** [comment]
@0x346e3730 I've looked into the usage of `CAP_NET_ADMIN` and although I see it's a requirement for the DHCP server, it also unlocks a number of other privileges such as being able to modify network interface configuration (full details here: https://man7.org/linux/man-pages/man7/capabilities.7.html), which we have some reluctance to allow. As we also don't see other requests for this, I'm inclided to keep this PR open and allow other users to upvote, and then we can revisit this feature request again later.

**Comment by BeauAgst** [comment]
This feature is required for anyone who is using Virgin Media in the UK, as you need to allow Pi-hole to take over DHCP. Without it, Pi-hole is entirely useless and does not work.

For context, Virgin Media has a 17% market share in the UK.

**Comment by unlimited01** [comment]
This one bit me as I was migrating my Pihole server to an Umbrel deploy.  I had to eventually roll back.
the open port (`udp67`) and `NET_ADMIN` is how I run my existing pi-hole container now.  If I can't have y pi-hole instance run my DHCP, it's a blocker for me.

**Comment by SolidRhino** [comment]
Please add this option. Ziggo (internet provider in the Netherlands) don't allow setting the DNS to an internal server.

**Comment by nmfretz** [comment]
Thanks @0x346e3730, we've decided to move forward with adding DHCP functionality to the app!

> I have not no as I have no test environment and cannot break my install

I will test this evening when others on my network aren't using the internet. I have merged in the most recent changes from Master and tested that the app does not break when updated, so if you want to test this you do not need to worry about your install breaking.

Here are the steps to update your app if you decide you want to test:

1) Connect to your Umbrel device by SSH on a Raspberry Pi or by accessing a terminal on your Linux machine.

 2) Run `sudo ~/umbrel/scripts/repo checkout https://github.com/0x346e3730/umbrel-apps.git#pihole-dhcp` to change the apps repo on your umbrel so that you can update the pi-hole version from this PR.
 
 3) Either update pi-hole through the app store in the browser UI, or by running the following command:
 `sudo ~/umbrel/scripts/app update pi-hole`
 
 After you are done testing you can reset the apps repo to the official umbrel-apps repo:
 `sudo ~/umbrel/scripts/repo checkout https://github.com/getumbrel/umbrel-apps.git`
 
 and restore the previous version of pi-hole through the app store in the browser UI, or by running:
 `sudo ~/umbrel/scripts/app update pi-hole`

**Comment by nmfretz** [comment]
Leaving this here for our reference:

[Docker DHCP docs](https://docs.pi-hole.net/docker/dhcp/#docker-dhcp-and-network-modes)

**Comment by 0x346e3730** [comment]
Since then I've bought a router so that even if I change my internet provider or my box I don't hade to re-do my whole install so I won't be able to test it neither but I'm glad it's getting added !

**Comment by nmfretz** [comment]
Going live! Thanks for the feedback everyone.

**Pull request merged**